engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CachingSimpletonActor.scala

```diff
@@ -0,0 +1,47 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import akka.actor.{Actor, ActorLogging, Props}
+import cromwell.core.JobOutputs
+import cromwell.database.CromwellDatabase
+import cromwell.database.sql.{CachedResult, MetaInfoId}
+import cromwell.engine.workflow.lifecycle.execution.callcaching.CachingSimpletonActor.{CachedOutputLookupSucceeded, CachedOutputLookupFailed, CacheResultResponse}
+import wdl4s.TaskOutput
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object CachingSimpletonActor {
+    def props(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput]): Props =
+        Props(new CachingSimpletonActor(metaInfoId, taskOutputs))
+
+  sealed trait CachedResultResponse
+  case class CachedOutputLookupFailed(metaInfoId: MetaInfoId, failure: Throwable) extends CachedResultResponse
+  case class CachedOutputLookupSucceeded(jobOutputs: JobOutputs) extends CachedResultResponse
+}
+
+
+class CachingSimpletonActor(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput]) extends Actor with ActorLogging with CromwellDatabase with CacheResultResponse {
+  {
+    implicit val ec: ExecutionContext = context.dispatcher
+
+    val callCache = new CallCache(CromwellDatabase.databaseInterface)
+    val replyTo = context.parent
+
+    callCache.fetchCachedResult(metaInfoId) onComplete {
+      case Success(Some(s)) => processJobOutputs(s, taskOutputs)
+      case Failure(t) => replyTo ! CachedOutputLookupFailed(metaInfoId, t)
+    }
+
+    def processJobOutputs(cachedResult: CachedResult, taskOutputs: Seq[TaskOutput]) = {
+      callCache.convertToJobOutputs(cachedResult, taskOutputs) match {
+        case (jobOutputs: JobOutputs) => replyTo ! CachedOutputLookupSucceeded(jobOutputs)
+        case (t: Throwable) => replyTo ! CachedOutputLookupFailed(metaInfoId, t)
```

Comment:
`convertToJobOutputs` can't ever return a Throwable. I think your maps, flatmaps and matches got jumbled up somewhere.

I had a go at refactoring 😳:
```
class CachingSimpletonActor(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput]) extends Actor with ActorLogging with CromwellDatabase {

  implicit val ec: ExecutionContext = context.dispatcher

  val callCache = new CallCache(CromwellDatabase.databaseInterface)

  def fetchCachedResult = callCache.fetchCachedResult(metaInfoId) flatMap {
    case Some(r) => Future.fromTry(Try(callCache.convertToJobOutputs(r, taskOutputs)))
    case None => Future.failed(new IllegalArgumentException("Bad job simpletons"))
  }

  val response: Future[CachedResultResponse] = fetchCachedResult map {
    CachedOutputLookupSucceeded(_)
  } recover { case t => CachedOutputLookupFailed(metaInfoId, t) }

  response.pipeTo(context.parent)

  override def receive: Receive = {
    case any => log.error("Unexpected message to CachingSimpletonActor: " + any)
  }
}
```